packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergWebViewActivity.java

```diff
@@ -204,25 +214,61 @@ public void onPageFinished(WebView view, String url) {
-                    evaluateJavaScript(preventAutosaves);
+                String injectGutenbergObserver = getFileContentFromAssets("gutenberg-web-single-block/gutenberg-observer.js");
+                evaluateJavaScript(injectGutenbergObserver);
+            }
+        });
+    }
 
-                    String insertBlock = getFileContentFromAssets("gutenberg-web-single-block/insert-block.js").replace("%@","%s");
-                    String blockContent = getIntent().getExtras().getString(ARG_BLOCK_CONTENT);
-                    insertBlock = String.format(insertBlock, blockContent);
-                    evaluateJavaScript(removeNewLines(insertBlock.replace("\\n", "\\\\n")));
+    private void evaluateJavaScript(String script) {
+        mWebView.evaluateJavascript(script, value ->
+                AppLog.e(AppLog.T.EDITOR, value));
+    }
 
-                    view.setVisibility(View.VISIBLE);
-                }, 2000);
-            }
+    private void onGutenbergReady() {
+        preventAutoSavesScript();
+        insertBlockScript();
+        final Handler handler = new Handler();
+        handler.postDelayed(() -> {
+            // We want to make sure that page is loaded
+            // with all elements before executing external JS
+            injectOnGutenbergReadyExternalSources();
+            mForegroundView.postDelayed(() -> mForegroundView.setVisibility(View.INVISIBLE), 500);
+        }, 2000);
+    }
 
-            private void evaluateJavaScript(String script) {
-                mWebView.evaluateJavascript(script, value ->
-                        AppLog.e(AppLog.T.EDITOR, value));
-            }
-        });
+    private void injectOnGutenbergReadyExternalSources() {
+        List<String> list = getOnGutenbergReadyExternalSources();
+        for (String file : list) {
+            evaluateJavaScript(file);
+        }
+    }
+
+    protected List<String> getOnGutenbergReadyExternalSources() {
+        return new ArrayList<>();
+    }
+
+    private void injectOnPageLoadExternalSources() {
+        List<String> list = getOnPageLoadExternalSources();
+        for (String file : list) {
+            evaluateJavaScript(file);
+        }
+    }
+
+    protected List<String> getOnPageLoadExternalSources() {
+        return new ArrayList<>();
+    }
+
+    private void preventAutoSavesScript() {
+        String preventAutosaves = getFileContentFromAssets("gutenberg-web-single-block/prevent-autosaves.js");
+        evaluateJavaScript(preventAutosaves);
+    }
+
+    private void insertBlockScript() {
+        String insertBlock = getFileContentFromAssets("gutenberg-web-single-block/insert-block.js").replace("%@","%s");
+        String blockContent = getIntent().getExtras().getString(ARG_BLOCK_CONTENT);
+        insertBlock = String.format(insertBlock, blockContent);
+        evaluateJavaScript(removeNewLines(insertBlock.replace("\\n", "\\\\n")));
```

Comment:
Not a big deal and you didn't add this line in this PR, so feel free to leave it as-is, but there's a fair bit happening on this line, and it might be helpful to either add a comment or extract out a descriptively-named variable to make it clearer why we're adding the extra escaping and then removing new lines.